apps/src/lib/node/ledger/shell/queries.rs

```diff
@@ -596,6 +621,66 @@ where
     fn get_current_decision_height(&self) -> BlockHeight {
         self.last_height + 1
     }
+
+    #[inline]
+    fn get_ethbridge_from_namada_addr(
+        &self,
+        validator: &Address,
+        epoch: Option<Epoch>,
+    ) -> Option<EthAddress> {
```

Comment:
same issue as above, it can return `None` if we are a fullnode.

Reply:
like we discussed in our call, the `Shell<...>` should take an extra parameter`NodeTag`  to distinguish between different node kinds, and only implement methods like `prepare_proposal` e.g. if we are a `NodeTag = Validator`. this would also obviate the need to error out from methods like `get_ethbridge_from_namada_addr`, unless we actually got some storage reading error.

Reply:
Yeah that sounds like a good approach, have opened https://github.com/anoma/namada/issues/521